core/new-gui/src/app/workspace/component/input-autocomplete/input-autocomplete.component.ts

```diff
@@ -0,0 +1,74 @@
+import { Component, OnInit } from "@angular/core";
+import { FieldType } from "@ngx-formly/core";
+import { UntilDestroy, untilDestroyed } from "@ngneat/until-destroy";
+import { UserFileService } from "src/app/dashboard/service/user-file/user-file.service";
+import { FormControl } from "@angular/forms";
+
+@UntilDestroy()
+@Component({
+  selector: "texera-input-autocomplete-template",
+  templateUrl: "./input-autocomplete.component.html",
+  styleUrls: ["input-autocomplete.component.scss"],
+})
+
+/* *
+ * The FieldType<any> is a workaround for the issue of not assignable FormControl.
+ * details https://github.com/ngx-formly/ngx-formly/issues/2842#issuecomment-1066116865
+ * need to upgrade formly to v6 to properly fix this issue.
+ */
+export class InputAutoCompleteComponent extends FieldType<any> implements OnInit {
+  inputValue: string = "";
+  title?: string;
+  // the autocomplete selection list
+  public selections: string[] = [];
+  fullList: string[] = [];
+
+  constructor(public userFileService: UserFileService) {
+    super();
+  }
+
+  // TODO: write this function because formcontrol is a nonnullable and read-only field. This is used to fit the test.
+  getControl() {
+    if (this.field == undefined) return new FormControl({});
+    return this.formControl;
+  }
+
+  ngOnInit() {
+    if (this.field) {
+      this.inputValue = this.field.formControl.value;
+      if (this.inputValue?.length > 0)
+        this.userFileService
+          .getAutoCompleteUserFileAccessList(encodeURIComponent(this.inputValue))
+          .pipe(untilDestroyed(this))
+          .subscribe(autocompleteList => {
+            this.selections = autocompleteList.concat();
```

Comment:
add a check to only update the selections if there is any difference. this can reduce the frequency of the dropdown menu being updated, thus less "blinking" on the UI.

Reply:
solved